specs/labels-and-localization/README.md

```diff
@@ -0,0 +1,188 @@
+- Define an MSBuild task that copies the resx file, similar to what's outlined in [this GitHub comment](https://github.com/ni/nimble/issues/558#issuecomment-1129279985). Not much better / still a manual process.
+
+### Plan for Client-Provided Labels
+Examples: Button content, menu item content, `nimble-banner` `dismissButtonLabel`, `nimble-table` column `action-menu-label`
+
+Almost all of these labels are context-specific, i.e. different buttons on the page will have different text, and different columns will have different action menu labels.
+
+Clients will localize those labels themselves:
+- For Angular, they can use the `localize` function and/or `i18n` attributes/ attribute prefixes
+- For Blazor, they can use `.resx` files and `IStringLocalizer`
+
+The banner's `dismissButtonLabel` will be semi-redundant once we have a Nimble-provided label for banner dismiss buttons, so we could consider removing that attribute in the future.
+
+## Alternative Implementations / Designs
+
+Alternative options for Nimble-provided labels:
+### Slots
+
+Create slots per localized string on the element.
+
+```html
+<nimble-table>
+   <span slot="collapse-all-button-label">Hello World</span>
+</nimble-table>
+```
+
+Pros:
+- Useful for mixed content (if that is even needed, maybe not?)
+- Can maybe make shareable slotted content within a framework
+  - Angular: Create a shared component that emits elements with those slots
+  - Blazor: I expect a blazor component written that way would work too
+- Slots can have default content easily
+
+Cons:
+- DOM gets littered with many additional elements for providing localized strings
+- Lots of duplicated slotted content / elements. Have to emit into every table element as children.
+
+### Attributes
+
+Create attribute per localized string on the element.
+
+```html
+<nimble-table
+    collapse-all-button-label="Hello World"
+></nimble-table>
+```
+
+Pros:
+- Easy discovery of available properties? Type checked in Angular at least
+
+Cons:
+- Maybe not as easy to share / reuse attribute configuration?
+    - Angular: Might need to write a custom directive that will emit the attributes onto the host. I think that works
+    - Blazor: ?
+- Lots of duplicated attributes. Have to emit into every table element as children.
+- Only text content (maybe that's okay)
+
+## Open Issues
+
+- Are the label IDs (`label-number-field-increment`) sufficient information for someone to translate the strings from? (Should we include an additional/optional description with each label/string?)
+- Does Nimble need to support more advanced use cases of strings/labels like format specifiers or [ICU expressions](https://angular.io/guide/i18n-common-prepare#icu-expressions)?
```

Comment:
I cover some possibly related use cases in #1268.  For table columns that display numbers formatted as strings with internationalization considerations, I am trying to rely on the browser to provide localized strings like month and day names. And those browser APIs have their own formatting APIs that don't require us to use format specifiers.

But I haven't found browser APIs for other strings like elapsed time ("minutes", "seconds") or units ("KB"). So the elapsed time and file size columns might want to leverage this infrastructure. At the moment I can't think of any new requirements that use case would place on this feature though. Mostly just sharing for awareness.